gateway/gateway.proto

```diff
@@ -47,6 +48,9 @@ service Gateway {
     // transaction function and return the result to the client. No ledger updates are made.
     // The gateway will select an appropriate peer to query based on block height and load.
     rpc Evaluate(EvaluateRequest) returns (EvaluateResponse);
+
+    // The ChaincodeEvents service supplies a stream of chaincode events emitted by the requested chaincode.
+    rpc ChaincodeEvents(SignedChaincodeEventsRequest) returns (stream ChaincodeEventsResponse);
```

Comment:
I believe that the events will be ordered here -- In the order of <blocknumber,txNumber>. If yes, this would need some changes in the commit notification hand-off between ledger and gateway, so this information is not lost. As far as this PR is concerned, including this aspect in the comment should suffice.

Reply:
Obtaining chaincode events using the ledger commit notification does not provide any ordering of transactions within a block so, for now, I'll just comment that responses are ordered by ascending block number.

Reply:
I believe that the ordering of events even within block should be important as it defines the commit order. We can always change the internal implementation for the desired behavior.